Assets/MixedRealityToolkit/_Core/Managers/MixedRealityManager.cs

```diff
@@ -955,33 +893,67 @@ private void GetComponentByType(Type type, out IMixedRealityManager manager)
         private void GetComponentByTypeAndName(Type type, string managerName, out IMixedRealityManager manager)
         {
             if (type == null) { throw new ArgumentNullException(nameof(type)); }
-            if (string.IsNullOrEmpty(managerName)) { throw new ArgumentNullException(nameof(managerName)); }
 
             manager = null;
 
             for (int i = 0; i < mixedRealityComponentsCount; i++)
             {
-                if (MixedRealityComponents[i].Item1.Name == type.Name &&
-                    MixedRealityComponents[i].Item2.Name == managerName)
+                if (CheckComponentMatch(type, managerName, MixedRealityComponents[i]))
                 {
                     manager = MixedRealityComponents[i].Item2;
                     break;
                 }
+            }
+        }
+
+        private void GetComponentsByType(Type type, ref List<IMixedRealityManager> managers)
```

Comment:
if we modify the list while we get a component, will that cause problems?

Reply:
The managers list being passed in? It's created in the original method call and not passed on until the return. I'm not sure where else it could be modified.